earth_enterprise/tutorial/SConstruct

```diff
@@ -34,30 +34,62 @@ if (not installdir) and ('install' in COMMAND_LINE_TARGETS):
 tutorials_dir = installdir + '/tutorial'
 root_dir = tutorials_dir + '/opt/google/share/tutorials/fusion/'
 
-def makeHardLinks(instdir, target_dir, excluded_list):
-    # get contents of target_dir and make hard links to files in instdir
+# Get the current directory, since SCons will move into the directory
+#   it's using, and FusionTutorial is expected below the SConscript file.
+tutorial_src_dir = os.sep.join((os.getcwd().split(os.sep) +
+                    ['FusionTutorial']))
+
+def makeHardLinks(install_dir, target_dir, excluded_list):
+    # get contents of target_dir and make hard links to files in install_dir
     # recurse down the directories
 
     contents = os.listdir(target_dir)
-    target_dir = target_dir + "/"
+    target_dir = target_dir + "/"	
     # Ensure the install dir exists at each step
-    commands.getstatusoutput("mkdir -p " + instdir)
+    commands.getstatusoutput("mkdir -p " + install_dir)
     for file in contents:
         if file in excluded_list:
             continue
         target_file = target_dir + file
         if os.path.isdir(target_file):
-            subdir = instdir + "/" + file
+            subdir = install_dir + "/" + file
             makeHardLinks(subdir, target_file,
                                              excluded_list)
         else:
             commands.getstatusoutput("ln \"" + target_file + "\" " +
-                                     instdir)
+                                     install_dir)
+
+
+def makeSoftLinks(dest_dir, src_dir, excluded_list):
+    """Make soft links of all files in src_dir and place
+    them in dest_dir. Don't link to files in excluded_list.
+    Recurse through all subdirectories of src_dir.
+    """
+    # Strip trailing slashes because they cause trouble later
+    dest_dir = dest_dir.rstrip(os.sep)
+    src_dir = src_dir.rstrip(os.sep)
+    src_contents = os.listdir(src_dir)
+    # Ensure the destination directory exists at each step
+    commands.getstatusoutput("mkdir -p " + dest_dir)
+    for file in src_contents:
```

Comment:
Did you consider creating a common function that performs this same loop for both hard and soft links?  The common function could be passed a closure or the link command to execute since that seems to be the only difference.  Or if the hard link method is dead then maybe we should delete it.

Reply:
I propose deleting the makeHardLinks function; I'm not sure hard links are required and soft links seem to work adequately. If a use case comes up in the future, the git history can be called upon for a hard link function and additional logic built to answer the question "which link type is best?". I suggest that logic is outside the scope of the original issue.

Reply:
No, since the issue just proposed either making soft links or copying, the logic of "hard links vs soft links" wasn't considered. It seems outside the scope of the issue.